pkg/endpoints/pipeline.go

```diff
@@ -126,27 +126,43 @@ func (r Resource) getPipelineImpl(name, namespace string) (v1alpha1.Pipeline, er
 	return *pipeline, nil
 }
 
-/* Get all pipeline runs in a given namespace */
+/* Get all pipeline runs in a given namespace, filters based on query parameters */
 func (r Resource) getAllPipelineRuns(request *restful.Request, response *restful.Response) {
+	// Request Params
 	namespace := request.PathParameter("namespace")
-	pipelineruns := r.PipelineClient.TektonV1alpha1().PipelineRuns(namespace)
+	// Query Params
 	repository := request.QueryParameter("repository")
+	name := request.QueryParameter("name")
+	logging.Log.Debugf("In getAllPipelineRuns: namespace: `%s`, repository query: `%s`, repository query: `%s`", namespace, repository, name)
 
-	logging.Log.Debugf("In getAllPipelineRuns: namespace: `%s`, repository query: `%s`", namespace, repository)
-
+	pipelinerunInterface := r.PipelineClient.TektonV1alpha1().PipelineRuns(namespace)
 	var pipelinerunList *v1alpha1.PipelineRunList
+	var labelSelector string // key1=value1,key2=value2, ...
 	var err error
+
+	// repository query filter
 	if repository != "" {
 		server, org, repo := getGitValues(repository)
-		match := gitServerLabel + "=" + server + "," + gitOrgLabel + "=" + org + "," + gitRepoLabel + "=" + repo
-		pipelinerunList, err = pipelineruns.List(metav1.ListOptions{LabelSelector: match})
-		logging.Log.Debugf("+%v", pipelinerunList.Items)
-	} else {
-		pipelinerunList, err = pipelineruns.List(metav1.ListOptions{})
-		logging.Log.Debugf("+%v", pipelinerunList.Items)
-		if err != nil {
-			utils.RespondError(response, err, http.StatusNotFound)
-			return
+		labels := []string {
+			gitServerLabel + "=" + server,
+			gitOrgLabel + "=" + org,
+			gitRepoLabel + "=" + repo,
+		}
+		labelSelector = strings.Join(labels,",")
+	}
+	pipelinerunList, err = pipelinerunInterface.List(metav1.ListOptions{LabelSelector: labelSelector})
+	logging.Log.Debugf("+%v", pipelinerunList.Items)
+	if err != nil {
+		utils.RespondError(response, err, http.StatusNotFound)
+		return
+	}
+	// name query filter
+	if name != "" {
+		for i := range pipelinerunList.Items {
+			pipelinerun := pipelinerunList.Items[i]
+			if pipelinerun.ObjectMeta.Name != name {
+				pipelinerunList.Items = append(pipelinerunList.Items[:i], pipelinerunList.Items[i+1:]...)
```

Comment:
Could be field selector?

Reply:
Had misunderstanding with field selector, realized this could be done. I implemented this, but it is not supported on the gotest side. I have been looking into contributing some changes on the fake testing side so that we can do this.

Reply:
What part isn't supported on the gotest sides, just using field selector?

If it works for our actual implementation code we should still use it here, and if it doesn't for tests I'm sure we can workaround it (carefully adding/removing runs and checking counts for example, or just stick to the for loop in your test code)